### PR TITLE
DDCE-2716 - Migrate reatcive mongo to hmrc-mongo + deps upgrade

### PR DESCRIPTION
**Pull request by frikit**

### CODE REVIEW

File: app/config/FrontendAppConfig.scala [hunk 1]
```diff
@@ -59,7 +52,8 @@ class FrontendAppConfig @Inject() (configuration: Configuration,
     "cymraeg" -> Lang(WELSH)
   )
 
-  def routeToSwitchLanguage: String => Call =
-    (lang: String) => routes.LanguageSwitchController.switchToLanguage(lang)
+  val cacheTtlSeconds: Int = configuration.get[Int]("mongodb.timeToLiveInSeconds")
```
Comment (anu9) [inline app/config/FrontendAppConfig.scala:55]: should be Long instead; Type-mismatch error

File: app/repositories/DefaultSessionRepository.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    collectionName = "user-answers",
+    mongoComponent = mongo,
+    domainFormat = Format(UserAnswers.reads, UserAnswers.writes),
+    indexes = Seq(
+      IndexModel(
+        ascending("updatedAt"),
+        IndexOptions()
+          .unique(false)
+          .name("user-answers-updated-at-index")
+          .expireAfter(config.cacheTtlSeconds, TimeUnit.SECONDS))),
```
Comment (anu9) [inline app/repositories/DefaultSessionRepository.scala:46]: Fix type mismatch error; cacheTtylSeconds to Long instead

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -1,45 +1,30 @@
...
+    "com.typesafe.play"           %% "play-test"                % PlayVersion.current,
+    "org.scalatestplus.play"      %% "scalatestplus-play"       % "5.1.0",
+    "uk.gov.hmrc.mongo"           %% "hmrc-mongo-test-play-28"  % "0.73.0",
     "org.scalatestplus"           %% "scalatestplus-scalacheck" % "3.1.0.0-RC2",
-    "org.scalacheck"              %% "scalacheck"            % "1.16.0",
-    "com.github.tomakehurst"      % "wiremock-standalone"    % "2.27.2",
-    "com.vladsch.flexmark"        % "flexmark-all"           % "0.62.2"
+    "org.jsoup"                   %  "jsoup"                    % "1.15.3",
+    "org.scalatest"               %% "scalatest"                % "3.2.14",
+    "org.scalatestplus"           %% "mockito-4-6"              % "3.2.14.0",
```
Comment (anu9) [inline project/AppDependencies.scala:22]: This can be replaced by "org.mockito"   %% "mockito-scala-scalatest" % "1.17.12"

Comment (frikit) [reply project/AppDependencies.scala:22]: I dont see any reason why to do that, could you elaborate why you decided this?

File: app/config/FrontendAppConfig.scala (hunk 1 above)
Comment (frikit) [reply app/config/FrontendAppConfig.scala:55]: I dont see any reasons to put it as long, int is more than enough you will never ever overflow int value here so why use a type which use more space to store value?

File: app/repositories/DefaultSessionRepository.scala (hunk 2 above)
Comment (frikit) [reply app/repositories/DefaultSessionRepository.scala:46]: I dont see any reasons to put it as long, int is more than enough you will never ever overflow int value here so why use a type which use more space to store value?

File: project/AppDependencies.scala (hunk 3 above)
Comment (anu9) [reply project/AppDependencies.scala:22]: I think its not best approach to stick to a single version "mockito-4-6" mainly when there is an option to be replaced

File: app/config/FrontendAppConfig.scala (hunk 1 above)
Comment (anu9) [reply app/config/FrontendAppConfig.scala:55]: Though tests working fine; Type mis-match error is occuring

Comment (frikit) [reply app/config/FrontendAppConfig.scala:55]: I didnt have this warnings/error in my idea so this is why I was asking about the change but looking into the lib itself it is expecting a long value so it amkes sense will fix it soon

File: app/repositories/DefaultSessionRepository.scala (hunk 2 above)
Comment (frikit) [reply app/repositories/DefaultSessionRepository.scala:46]: will be fixed by prev  comment

File: app/config/FrontendAppConfig.scala (hunk 1 above)
Comment (frikit) [reply app/config/FrontendAppConfig.scala]: done

Comment (frikit) [reply app/config/FrontendAppConfig.scala]: done

File: project/AppDependencies.scala (hunk 3 above)
Comment (frikit) [reply project/AppDependencies.scala]: done good change 👍

**Review by anu9 — APPROVED** [review]
lgtm

**Pull request merged**